app/src/main/java/ch/blinkenlights/android/vanilla/RemoteControlImplLp.java

```diff
@@ -132,25 +137,34 @@ public void updateRemote(Song song, int state, boolean keepPaused) {
 			mShowCover = settings.getBoolean(PrefKeys.COVER_ON_LOCKSCREEN, PrefDefaults.COVER_ON_LOCKSCREEN) ? 1 : 0;
 		}
 
+		PlaybackService service = PlaybackService.get(null);
+
 		if (song != null) {
 			Bitmap bitmap = null;
 			if (mShowCover == 1 && (isPlaying || keepPaused)) {
 				bitmap = song.getCover(mContext);
 			}
 
-			session.setMetadata(new MediaMetadata.Builder()
+			MediaMetadata.Builder metadataBuilder = new MediaMetadata.Builder()
 				.putString(MediaMetadata.METADATA_KEY_ARTIST, song.artist)
 				.putString(MediaMetadata.METADATA_KEY_ALBUM, song.album)
 				.putString(MediaMetadata.METADATA_KEY_TITLE, song.title)
 				.putLong(MediaMetadata.METADATA_KEY_DURATION, song.duration)
-				.putBitmap(MediaMetadata.METADATA_KEY_ALBUM_ART, bitmap)
-			.build());
+				.putBitmap(MediaMetadata.METADATA_KEY_ALBUM_ART, bitmap);
+
+			// logic copied from FullPlaybackActivity.updateQueuePosition()
+			if (PlaybackService.finishAction(service.getState()) != SongTimeline.FINISH_RANDOM) {
+				metadataBuilder.putLong(MediaMetadata.METADATA_KEY_TRACK_NUMBER, service.getTimelinePosition() + 1);
+				metadataBuilder.putLong(MediaMetadata.METADATA_KEY_NUM_TRACKS, service.getTimelineLength());
+			}
+
+			session.setMetadata(metadataBuilder.build());
 		}
 
 		int playbackState = (isPlaying ? PlaybackState.STATE_PLAYING : PlaybackState.STATE_PAUSED);
 
 		session.setPlaybackState(new PlaybackState.Builder()
-			.setState(playbackState, PlaybackState.PLAYBACK_POSITION_UNKNOWN , 1.0f)
+			.setState(playbackState, service.getPosition(), 1.0f)
```

Comment:
So this gets updated if a song changes or if the user seeks?

Where is this visible and why is this better than not reporting the position?

Reply:
This is visible in any Bluetooth remote controller that shows playback progress, for instance, all my Windows and cars AVRCP clients.